packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts

```diff
@@ -458,6 +444,33 @@ export class TaskDefinition extends TaskDefinitionBase {
       throw new Error('Cannot specify runtimePlatform in non-Fargate compatible tasks');
     }
 
+    //FARGATE compatible tasks pre-checks
+    if (this.isFargateCompatible) {
+      if (this.networkMode !== NetworkMode.AWS_VPC) {
+        throw new Error(`Fargate tasks can only have AwsVpc network mode, got: ${this.networkMode}`);
+      }
+
+      if (props.placementConstraints && props.placementConstraints.length > 0) {
+        throw new Error('Cannot set placement constraints on tasks that run on Fargate');
+      }
+
+      if (!props.cpu || !props.memoryMiB) {
+        throw new Error(`Fargate-compatible tasks require both CPU (${props.cpu}) and memory (${props.memoryMiB}) specifications`);
+      }
+
+      if (props.inferenceAccelerators && props.inferenceAccelerators.length > 0) {
+        throw new Error('Cannot use inference accelerators on tasks that run on Fargate');
+      }
+
+      // Validate the CPU and memory size for the Windows operation system family.
+      if (props.runtimePlatform?.operatingSystemFamily?.isWindows()) {
+        this.checkFargateWindowsBasedTasksSize(props.cpu, props.memoryMiB, props.runtimePlatform);
+      }
+
+      // Check the combination as per doc https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html
+      this.validateFargateTaskDefinitionMemoryCpu(props.cpu, props.memoryMiB);
```

Comment:
```suggestion
     this.node.addValidation({ 
       validate: () => this.validateFargateTaskDefinitionMemoryCpu(props.cpu, props.memoryMiB)
     });
```
Please note https://github.com/aws/aws-cdk/pull/30166/files#r1641289793

Reply:
I have made all the suggested changes @lpizzinidev

Reply:
Please take a look once